### PR TITLE
For k8s registry the instance handler is redudant

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/multicluster.go [hunk 1]
```diff
@@ -130,8 +130,9 @@ func (m *Multicluster) AddMemberCluster(clients kubelib.Client, clusterID string
 	m.remoteKubeControllers[clusterID] = &remoteKubeController
 	m.m.Unlock()
 
+	// Only need to add service handler for kubernetes registry as `initRegistryEventHandlers`,
+	// because when endpoints update `XDSUpdater.EDSUpdate` has already been called.
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/kube/controller/multicluster.go:134]: I guess this was added for workloads selecting pods to reconcile foreign registry instances. @rshriram ?

Comment (rshriram) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go:134]: Oh yes.. please dont remove this.. This is for workload entry

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go:134]: nope, foreign service instance is not supported yet in multi-cluster model.

Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go:134]: I see. This is only for multi cluster. Missed that part.

Comment (costinm) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go:134]: Ideally all endpoint changes would use EDSUpdate - even for WorkloadEntry/ServiceEntry

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go:134]: Yeah, and after #25502 it is possible

**Pull request merged**